services/catarse.js/legacy/src/root/blog-banner.tsx

```diff
@@ -0,0 +1,76 @@
+import m from 'mithril';
+import prop from 'mithril/stream';
+import _ from 'underscore';
+import h from '../h';
+import blogVM from '../vms/blog-vm';
+
+export default class BlogBanner implements m.Component<{}> {
+    oninit(vnode) {
+        const posts = prop([])
+        const error = prop(false)
+
+        async function loadPosts() {
+            try {
+                const loadedPosts = await blogVM.getBlogPosts()
+                posts(loadedPosts)
+            } catch (e) {
+                console.log('BlogBanner error', e)
+                error(e)
+            } finally {
+                h.redraw()
+            }
+        }
+        loadPosts()
+        vnode.state.posts = posts;
+        vnode.state.error = error;
+    }
+
+    view({state}) {
+        const posts = state.posts() || []
+        const hasError = state.error() || false
+        return (
+            <section class="section-large bg-gray before-footer" id="blog">
+                <div class="w-container">
+                    <div class="u-text-center">
+                        <a href="http://blog.catarse.me" target="blank">
+                            <img src="/assets/icon-blog.png" alt="Icon blog" class="u-marginbottom-10"/>
+                        </a>
+                        <div class="fontsize-large u-marginbottom-60 text-success">
+                            <a href="http://blog.catarse.me" class="link-hidden-success" target="__blank">
+                                Blog do Catarse
+                            </a>
+                        </div>
+                    </div>
+                    <div class="w-row">
+                        {
+                            posts.map(post => {
```

Comment:
**Sugestão:**
Aqui há a oportunidade de transformar isso num componente próprio de `BlogPostBanner`. Ou se achar exagero, pelo menos separar essa lógica do html, criando uma função ali em cima do return, se for possível.

Reply:
Blz, eu vou colocar mesmo. Melhor aumentar o padrão mesmo.

Reply:
Massa @grpse .